### PR TITLE
Bluetooth: controller: Improve pdu_adv structures

### PR DESCRIPTION
**Pull request by andrzej-kaczmarek**
Structures and members are renamed to make it more semantically correct.
Structure of common extended advertising payload was extended to allow more direct access to extended header contents.

I also changes code in few places to make use of new extended struct, probably there are some more places to update although they are not obvious at first glance.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:925]: There is some changes to the implementation here, hence blocking this PR until #29778 is merged. Otherwise changes proposed here looks good.

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by carlescufi** [comment]
@andrzej-kaczmarek this needs a rebase and addressing the review comments

File: subsys/bluetooth/controller/ll_sw/ull_scan_aux.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_scan_aux.c:137]: Change other places from `ptr = (uint8_t *)h + sizeof(*h);` to `ptr = h->data;`

File: subsys/bluetooth/controller/hci/hci.c
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:3828]: Change all places from `ptr = (uint8_t *)h + sizeof(*h);` to `ptr = h->data;` ?

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:514]: You missed this one?

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by cvinayak — APPROVED** [review]

**Comment by cvinayak** [comment]
@wopu-ot @thoh-ot @asmk-ot  This one needs another reviewer... estimating a max 1 hour inspection effort. As there is no functional change and CI is green with the babblesim tests passing (if has been run!), please try to approve this PR today.

**Review by wopu-ot — APPROVED** [review]

**Comment by carlescufi** [comment]
@andrzej-kaczmarek please rebase

**Comment by carlescufi** [comment]
@andrzej-kaczmarek please rebase

**Comment by andrzej-kaczmarek** [comment]
@carlescufi rebased

**Pull request merged**
